### PR TITLE
usm: http: tests: Unify tests

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Unifies different HTTP tests and reduce duplication and redundant wrappers

### Motivation

Cleanup

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/usm/monitor_test.go [hunk 1]
```diff
@@ -402,9 +402,21 @@ func (s *HTTPTestSuite) TestHTTPMonitorIntegration() {
...
 				t.Run(testNameHelper("with keep alive", "without keep alive", keepAliveEnabled), func(t *testing.T) {
-					testHTTPMonitor(t, tt.targetAddress, tt.serverAddress, 100, testutil.Options{
-						EnableKeepAlive: keepAliveEnabled,
-					})
+					monitor := newHTTPMonitor(t)
+
+					srvDoneFn := testutil.HTTPServer(t, tt.serverAddress, testutil.Options{EnableKeepAlive: keepAliveEnabled})
+					t.Cleanup(srvDoneFn)
+
+					// Perform a number of random requests
```
Comment (amitslavin) [inline pkg/network/usm/monitor_test.go:362]: lets make the doc more understandable, that the random is the content of the requests.

File: pkg/network/usm/monitor_test.go [hunk 2]
```diff
@@ -583,30 +579,16 @@ func requestGenerator(t *testing.T, targetAddr string, reqBody []byte) func() *n
...
-func requestNotIncluded(t *testing.T, allStats map[http.Key]*http.RequestStats, req *nethttp.Request) {
+func checkRequestIncluded(t *testing.T, allStats map[http.Key]*http.RequestStats, req *nethttp.Request, expectedToBeIncluded bool) {
 	included, err := isRequestIncludedOnce(allStats, req)
 	require.NoError(t, err)
-	if included {
-		expectedStatus := testutil.StatusFromPath(req.URL.Path)
+	if included != expectedToBeIncluded {
 		t.Errorf(
-			"should not find HTTP transaction matching the following criteria:\n path=%s method=%s status=%d",
+			"%s not find HTTP transaction matching the following criteria:\n path=%s method=%s status=%d",
```
Comment (amitslavin) [inline pkg/network/usm/monitor_test.go:587]: the begging of the error message does not sounds right.

**Review by amitslavin — CHANGES_REQUESTED** [review]

**Comment by amitslavin** [comment]
General comment: I do think that this test file should be much better documented

File: pkg/network/usm/monitor_test.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/usm/monitor_test.go:587]: talked by heart

File: pkg/network/usm/monitor_test.go [hunk 3]
```diff
@@ -326,7 +328,13 @@ func testNameHelper(optionTrue, optionFalse string, value bool) string {
 	return optionFalse
 }
 
-func (s *HTTPTestSuite) TestHTTPMonitorIntegration() {
+// TestSanity checks that USM capture a random generated 100 requests send to a local HTTP server under the following
+// conditions:
+// 1. Server and client support keep alive, and there is no NAT.
+// 1. Server and client do not support keep alive, and there is no NAT.
```
Comment (amitslavin) [inline pkg/network/usm/monitor_test.go:334]: fix the count

**Review by amitslavin — APPROVED** [review]

**Pull request merged**